### DEG analysis based on RNA-seq samples ###
### Boas Pucker ###
### b.pucker@tu-braunschweig.de ###


# --- loading DESeq --- #
library("DESeq2") 

# --- loading sampleTable (generated by Python script) --- #
csvfile <- "clean_sample_table.txt"
sampleTable <- read.csv(csvfile, row.names=1, sep="\t")
sampleTable$tissue <- as.factor( sampleTable$tissue )
tissue <- sampleTable$tissue
summary(sampleTable)

# --- loading the data matrix --- #
count_data_file <- "clean_data_matrix.txt"
countdata <- read.csv(count_data_file,row.names=1, header=T, sep="\t")
summary(countdata)

# --- set working dir where all files will be saved later --- #
"working dir set to:"
(working_dir = "xxx")
setwd( working_dir  )


# -- removal of not or low expressed genes --- #
# Current option: removal of genes if less than 50 percent of all samples show expression above cutoff

thres = 2
nzIndex = as.vector(which(apply(countdata,1,function(x){sum(x>thres)/length(x)})>=0.5))
head(nzIndex)

countdata1 = countdata[nzIndex,]
"number of genes after eliminating low expressed genes:"
nrow(countdata1)

"amount of eliminated genes:"
(nr_eliminated_genes <- nrow(countdata) - nrow(countdata1))

# --- construction of DESeqDataSet --- #
dds <- DESeqDataSetFromMatrix( countData=countdata1, colData=sampleTable, design = ~ tissue )
"number of genes in matrix:"
nrow(dds)


# --- differential expression analysis --- #
dds <- DESeq(dds)
resultsNames(dds)


# --- print session information --- #
sessionInfo()
